port/unix/omrmmap.c

```diff
@@ -187,8 +285,18 @@ omrmmap_unmap_file(struct OMRPortLibrary *portLibrary, J9MmapHandle *handle)
 
 	if (handle != NULL) {
 		Trc_PRT_mmap_unmap_file_unix_values(handle->pointer, handle->size);
+
+#if defined(J9ZOS390)
+		if (NULL != handle->allocPointer) {
+			portLibrary->mem_free_memory(portLibrary, handle->allocPointer);
+		} else {
+			rc = munmap(handle->pointer, handle->size);
+			omrmem_categories_decrement_counters(handle->category, handle->size);
+		}
+#else /* defined(J9ZOS390) */
 		rc = munmap(handle->pointer, handle->size);
 		omrmem_categories_decrement_counters(handle->category, handle->size);
+#endif /* defined(J9ZOS390) */
```

Comment:
It might be preferable to avoid repeating the `else` block:
```c
#if defined(J9ZOS390)
		if (NULL != handle->allocPointer) {
			portLibrary->mem_free_memory(portLibrary, handle->allocPointer);
		} else
#endif /* defined(J9ZOS390) */
		{
			rc = munmap(handle->pointer, handle->size);
			omrmem_categories_decrement_counters(handle->category, handle->size);
		}
```

Reply:
Fixed.